tests/model_serving/model_runtime/mlserver/basic_model_deployment/test_huggingface_basic_model_deployment.py

```diff
@@ -0,0 +1,130 @@
+
+import pytest
+from simple_logger.logger import get_logger
+
+from ocp_resources.inference_service import InferenceService
+from ocp_resources.pod import Pod
+
+from utilities.constants import Protocols
+
+from tests.model_serving.model_runtime.mlserver.constant import (
+    BASE_RAW_DEPLOYMENT_CONFIG,
+    BASE_SERVERLESS_DEPLOYMENT_CONFIG,
+    MODEL_PATH_PREFIX,
+    HUGGING_FACE_GRPC_INPUT_QUERY,
+    HUGGING_FACE_REST_INPUT_QUERY,
+    HUGGING_FACE_FRAMEWORK,
+    NON_DETERMINISTIC_OUTPUT,
+)
+from tests.model_serving.model_runtime.mlserver.utils import validate_inference_request
+
+
+LOGGER = get_logger(name=__name__)
+
+MODEL_NAME: str = "distilgpt2"
+
+MODEL_VERSION: str = "v0.1.0"
+
+MODEL_NAME_DICT: dict[str, str] = {"name": MODEL_NAME}
+
+MODEL_STORAGE_URI_DICT: dict[str, str] = {"model-dir": f"{MODEL_PATH_PREFIX}/huggingface"}
+
+
+pytestmark = pytest.mark.usefixtures(
+    "root_dir", "valid_aws_config", "mlserver_rest_serving_runtime_template", "mlserver_grpc_serving_runtime_template"
+)
+
+
+@pytest.mark.parametrize(
+    ("protocol", "model_namespace", "mlserver_inference_service", "s3_models_storage_uri", "mlserver_serving_runtime"),
+    [
+        pytest.param(
+            {"protocol_type": Protocols.REST},
+            {"name": "huggingface-raw-rest"},
+            {
+                **BASE_RAW_DEPLOYMENT_CONFIG,
+                **MODEL_NAME_DICT,
+            },
+            MODEL_STORAGE_URI_DICT,
+            BASE_RAW_DEPLOYMENT_CONFIG,
+            id="huggingface-raw-rest-deployment",
+        ),
+        pytest.param(
+            {"protocol_type": Protocols.GRPC},
+            {"name": "huggingface-raw-grpc"},
+            {
+                **BASE_RAW_DEPLOYMENT_CONFIG,
+                **MODEL_NAME_DICT,
+            },
+            MODEL_STORAGE_URI_DICT,
+            BASE_RAW_DEPLOYMENT_CONFIG,
+            id="huggingface-raw-grpc-deployment",
+        ),
+        pytest.param(
+            {"protocol_type": Protocols.REST},
+            {"name": "huggingface-serverless-rest"},
+            {**BASE_SERVERLESS_DEPLOYMENT_CONFIG, **MODEL_NAME_DICT},
+            MODEL_STORAGE_URI_DICT,
+            BASE_SERVERLESS_DEPLOYMENT_CONFIG,
+            id="huggingface-serverless-rest-deployment",
+        ),
+        pytest.param(
+            {"protocol_type": Protocols.GRPC},
+            {"name": "huggingface-serverless-grpc"},
+            {**BASE_SERVERLESS_DEPLOYMENT_CONFIG, **MODEL_NAME_DICT},
+            MODEL_STORAGE_URI_DICT,
+            BASE_SERVERLESS_DEPLOYMENT_CONFIG,
+            id="huggingface-serverless-grpc-deployment",
+        ),
+    ],
+    indirect=True,
+)
+class TestHuggingFaceModel:
+    """
+    Test class for huggingface model inference with MLServer runtime.
+
+    Covers multiple deployment scenarios:
+    - REST and gRPC protocols
+    - Raw and serverless deployment modes
+    - Response validation against snapshots
+    """
+
+    def test_huggingface_model_inference(
```

Comment:
Your tests and validations are exactly same. Why you need multiple duplicate ones? Just use parameterization for HF, lightgbm etc.